docs/configuration.rst

```diff
@@ -248,31 +262,45 @@ These settings can be used to specify where to find other settings through custo
     Path where to find `permissions.cfg`_ file. Can also be a directory path, where all contained configuration files
     will be parsed for ``permissions`` section and will be loaded sequentially.
 
+    Please refer to `permissions.cfg`_ for specific format details of the various parameters.
+
     .. note::
-        If a directory path is specified, the order of loaded configuration files is not guaranteed
-        (depending on OS implementation). Therefore, cross-file references to services or resources should be avoided
-        to ensure that, for example, any parent resource dependency won't be missing because it was specified in a
-        second file loaded after the first. Corresponding references can be duplicated across files and these conflicts
-        will be correctly handled according to configuration loading methodology.
-        Please refer to `permissions.cfg`_ for specific format details and loading methodology according to arguments.
+        If a directory path is specified, the order of loaded configuration files is alphabetical.
+
+    .. versionchanged:: 1.7.4
+
+        Loading order of multiple files was **NOT** guaranteed prior to this version.
+
+        With older versions, cross-file references to services or resources should be avoided to ensure that,
+        for example, any parent resource dependency won't be missing because it was specified in a
+        second file loaded after the first. Corresponding references can be duplicated across files
+        and these conflicts will be correctly handled according to configuration loading methodology.
+        Later versions are safe to assume alphabetical loading order.
 
 .. envvar:: MAGPIE_CONFIG_PATH
 
     Path where to find a combined YAML configuration file which can include ``providers``, ``permissions``, ``users``
     and ``groups`` sections to sequentially process registration or removal of items at `Magpie` startup.
 
-    .. note::
-        When provided, all other combinations of :envvar:`MAGPIE_CONFIG_DIR`, :envvar:`MAGPIE_PERMISSIONS_CONFIG_PATH`
-        and :envvar:`MAGPIE_PROVIDERS_CONFIG_PATH` are effectively ignored in favour of definitions in this file.
-        See :ref:`config_file` for further details and example.
+    See :ref:`config_file` for further details and an example of its structure.
+
+    .. versionchanged:: 3.6
+        The configuration can also contain a ``webhooks`` section, as described in :ref:`config_webhook` and
+        presented in the sample :ref:`config_file`.
+
+    .. warning::
+        When this setting is defined, all other combinations of :envvar:`MAGPIE_CONFIG_DIR`,
+        :envvar:`MAGPIE_PERMISSIONS_CONFIG_PATH` and :envvar:`MAGPIE_PROVIDERS_CONFIG_PATH` are effectively
+        ignored in favour of definitions in this file. It is not possible to employ the :ref:`config_file` at
+        the same time as multi-configuration loading from a directory.
```

Comment:
Haven't looked but I guess that there is a check to know if user tries to load `config_file` and multi-configuration at the same time?

Reply:
As soon as you define `MAGPIE_CONFIG_PATH`, you enforce the single-file mode. 
Otherwise, it attempts to resolve according to `MAGPIE_CONFIG_DIR`, `MAGPIE_PERMISSIONS_CONFIG_PATH` and `MAGPIE_PROVIDERS_CONFIG_PATH`. 

It is more detailed in their specific `:envvar:` section because it is repeating itself a lot otherwise and harder to maintain the docs accurate everywhere. I don't re-read everything all times, so I try to reference existing explanations instead of repeating to avoid discrepancies.

Reply:
Ok then if it's specified in the envvar part no need to repeat over here.
The goal was to avoid confusion in case someone tries both config setups and ends up with the `config_file` one.
But since it seems specified in envvar, that looks good to me.